### PR TITLE
Automatedscript

### PR DESCRIPTION
**Pull request by leslie-huang**
automate spell check, update version num, update NEWS, update cran-comments

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# leaflet.providers 0.1.0
+
+* Added a `NEWS.md` file to track changes to the package.
```
Comment (schloerke) [inline NEWS.md]: "Initialized package"

File: README.md [hunk 2]
```diff
@@ -6,22 +6,28 @@
 [![Codecov test coverage](https://codecov.io/gh/leslie-huang/leaflet.providers/branch/master/graph/badge.svg)](https://codecov.io/gh/leslie-huang/leaflet.providers?branch=master)
 <!-- badges: end -->
 
-The goal of leaflet.providers is to ...
+The goal of leaflet.providers is to provide regularly updated data on the third-party tile providers supported by `leaflet`. This package is to be used with `leaflet`.
 
 ## Installation
 
-You can install the released version of leaflet.providers from [CRAN](https://CRAN.R-project.org) with:
+You can install the released version of leaflet.providers from [CRAN](https://CRAN.R-project.org), or the development version for Github:
```
Comment (schloerke) [inline README.md]: no comma

File: README.md [hunk 3]
```diff
@@ -6,22 +6,28 @@
...
 ## Example
 
-This is a basic example which shows you how to solve a common problem:
+This is a basic example of the data that is automatically loaded with `leaflet.providers`.
 
 ``` r
 library(leaflet.providers)
-## basic example code
+## View supported tile providers
+names(providers)
```
Comment (schloerke) [inline README.md]: maybe add str of providers or provider.names

Comment (schloerke) [reply README.md]: This would work really well with a README.Rmd

File: scripts/update_data_and_pkg.R [hunk 4]
```diff
@@ -0,0 +1,64 @@
+# Increment release num.
+usethis::use_version()
+
+ct <- V8::v8()
```
Comment (schloerke) [inline scripts/update_data_and_pkg.R]: use the package to get "today's providers"

File: scripts/update_data_and_pkg.R [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+# Automate pkg components
+
+spelling::spell_check_test(vignettes = TRUE, error = FALSE,
+                           skip_on_cran = TRUE)
+
+# Auto update NEWS.md
+lines <- readLines("NEWS.md")
+filecon <- file("NEWS.md", "wt")
+writeLines(lines[1], con = filecon) # heading already creating by use_version
```
Comment (schloerke) [inline scripts/update_data_and_pkg.R]: "created"

File: scripts/update_data_and_pkg.R [hunk 6]
```diff
@@ -0,0 +1,64 @@
+# Increment release num.
+usethis::use_version()
```
Comment (schloerke) [inline scripts/update_data_and_pkg.R]: use `desc` package instead to be able to set the version directly

File: scripts/update_data_and_pkg.R [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+new_lines <- paste("#", Sys.Date(),
+                   "\n\n",
+                   "This submission updates `providers` and `providers.details`.",
+                   "\n",
+                   sep = " ")
+writeLines(new_lines, con = filecon)
+writeLines(lines, con = filecon)
+close(filecon)
+
+revdepcheck::revdep_check()
```
Comment (schloerke) [inline scripts/update_data_and_pkg.R]: add `revdepcheck::revdep_report_cran()`

Comment (schloerke) [reply scripts/update_data_and_pkg.R]: append it to the cran comments

File: scripts/update_data_and_pkg.R [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+filecon <- file("cran-comments.md", "wt")
+new_lines <- paste("#", Sys.Date(),
+                   "\n\n",
+                   "This submission updates `providers` and `providers.details`.",
+                   "\n",
+                   sep = " ")
+writeLines(new_lines, con = filecon)
+writeLines(lines, con = filecon)
+close(filecon)
+
```
Comment (schloerke) [inline scripts/update_data_and_pkg.R]: add in a `devtools::check()`

Comment (schloerke) [reply scripts/update_data_and_pkg.R]: add in a `rhub::check_for_cran()`

Comment (schloerke) [reply scripts/update_data_and_pkg.R]: add in a 
`devtools::check_win_oldrelease`
`devtools::check_win_release`
`devtools::check_win_devel`

Comment (schloerke) [reply scripts/update_data_and_pkg.R]: maybe call `rhub` after the `check_win_*` calls

File: scripts/update_data_and_pkg.R [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+providers <- purrr::map(names(providers.details), function(provider) {
+  if (is.null(variants[[provider]])) {
+    provider
+  } else {
+    c(provider, stringr::str_c(provider,'.',variants[[provider]]))
+  }
+}) %>% purrr::flatten_chr()
+
+providers <- setNames(as.list(providers), providers)
+
```
Comment (schloerke) [inline scripts/update_data_and_pkg.R]: insert tests here

File: README.Rmd [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+
+The goal of leaflet.providers is to provide regularly updated data on the third-party tile providers supported by `leaflet`. This package is to be used with `leaflet`.
+
+## Installation
+
+You can install the released version of leaflet.providers from [CRAN](https://CRAN.R-project.org):
+
+``` r
+# CRAN version
+<!-- install.packages("leaflet.providers") -->
```
Comment (schloerke) [inline README.Rmd]: technically this is in an R code bock... so I don't know if it'll work as a comment.  But I like the comment idea

File: README.Rmd [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+[![Travis build status](https://travis-ci.org/leslie-huang/leaflet.providers.svg?branch=master)](https://travis-ci.org/leslie-huang/leaflet.providers)
+[![Codecov test coverage](https://codecov.io/gh/leslie-huang/leaflet.providers/branch/master/graph/badge.svg)](https://codecov.io/gh/leslie-huang/leaflet.providers?branch=master)
+
+<!-- badges: end -->
+
+The goal of leaflet.providers is to provide regularly updated data on the third-party tile providers supported by `leaflet`. This package is to be used with `leaflet`.
+
+## Installation
+
+You can install the released version of leaflet.providers from [CRAN](https://CRAN.R-project.org):
```
Comment (schloerke) [inline README.Rmd]: add backticks for `leaflet.providers`

File: README.Rmd [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+## Example
+
+This is a basic example of the data that is automatically loaded with `leaflet.providers`.
+
+### View supported tile providers
+
+```{r example}
+library(leaflet.providers)
+str(providers)
+
```
Comment (schloerke) [inline README.Rmd]: remove this line

**Comment by leslie-huang** [comment]
Great! The movers showed up early but I’ll make these changes when I’m done dealing with them. I can just merge the PR after that (without another review)?

**Pull request merged**